WikidataGame/WikidataGame.Backend/Controllers/GamesController.cs

```diff
@@ -37,37 +37,43 @@ public class GamesController : CustomControllerBase
         /// <summary>
         /// Creates a new game and matches the player with an opponent
         /// </summary>
-        /// <param name="mapWidth">Width of generated map</param>
-        /// <param name="mapHeight">Height of generated map</param>
-        /// <param name="accessibleTilesCount">How many accessible tiles the generated map should contain.</param>
         /// <returns>Info about the newly created game</returns>
         [HttpPost]
         [ProducesResponseType(typeof(GameInfo), StatusCodes.Status200OK)]
-        public async Task<ActionResult<GameInfo>> CreateNewGame(
-            int mapWidth = GameConstants.DefaultMapWidth,
-            int mapHeight = GameConstants.DefaultMapHeight,
-            int accessibleTilesCount = GameConstants.DefaultAccessibleTilesCount)
+        public async Task<ActionResult<GameInfo>> CreateNewGame()
         {
             var user = await GetCurrentUserAsync();            
-            var game = await _gameRepo.RunningGameForPlayerAsync(user);
-            if (game == default(Models.Game))
-            {
-                game = await _gameRepo.GetOpenGameAsync();
-                if (game == default(Models.Game))
-                {
-                    game = await _gameRepo.CreateNewGameAsync(user, mapWidth, mapHeight, accessibleTilesCount);
-                }
-                else
-                {
-                    _gameRepo.JoinGame(game, user);
-                }
+            var openGames = await _gameRepo.GetOpenGamesAsync();
 
-                await _dataContext.SaveChangesAsync();
+            Models.Game game;
+            if (openGames.Count() <= 0 || openGames.All(g => g.GameUsers.SingleOrDefault(gu => gu.UserId == user.Id) != null)) 
```

Comment:
Minor nitpick: `openGames.Count()` can not be `< 0`. Otherwise looks good, nice to see that the change required for this was so small!